src/textual/dom.py

```diff
@@ -213,6 +218,8 @@ def __init__(
             dict[str, tuple[MessagePump, Reactive | object]] | None
         ) = None
         self._pruning = False
+        self._query_one_cache: LRUCache[tuple[object, ...], DOMNode] = LRUCache(1024)
```

Comment:
Do you think we could make the typing here more robust by writing out the types that are actually used and including the tuple length?

If the cache key changes it's handy to have the type checker point you to all the places where the code needs to change.